aio/aio-proxy/aio_proxy/response/build_response.py

```diff
@@ -5,31 +5,33 @@
 from aio_proxy.decorators.http_exception import http_exception_handler
 from aio_proxy.response.format_response import format_response
 from aio_proxy.response.format_search_results import format_search_results
-from aio_proxy.search.index import ElasticsearchSireneIndex
+from aio_proxy.search.es_search_runner import ElasticSearchRunner
 from aiohttp import web
 from sentry_sdk import capture_exception, push_scope
 
 
 @http_exception_handler
 def api_response(
-    request, extract_function: Callable, search_function: Callable
+    request,
+    extract_function: Callable,
+    search_type: str,
 ) -> dict[str, int]:
     """Create and format API response.
 
     Args:
         request: HTTP request.
         extract_function (Callable): function used to extract parameters.
-        search_function (Callable): function used for search .
+        search_type: type of search (text or geo).
     Returns:
         response in json format (results, total_results, page, per_page,
         total_pages)
     """
     try:
         search_params = extract_function(request)
-        search_results = search_function(ElasticsearchSireneIndex, search_params)
-        total_results = search_results["total_results"]
-        results = search_results["response"]
-        execution_time = search_results["execution_time"]
+        es_search_results = ElasticSearchRunner(search_params, search_type)
+        total_results = es_search_results.total_results
+        results = es_search_results.es_search_results
+        execution_time = es_search_results.execution_time
         results_formatted = format_search_results(results, search_params)
         response_formatted = format_response(
```

Comment:
Why re-declare scoped variables like `total_results and execution_time` ? You might pass around `es_search_results`

Reply:
Coming soon!